# DE Analysis of ORP Assembled Transcriptomes

To run a DE analysis in R with DESeq, you need a count matrix. A count matrix, refers to a table-like data structure that contains read counts for each gene or feature across different samples or conditions.Each row of the matrix represents a gene or feature, and each column represents a sample.

The count matrix is generated by aligning sequencing reads to a reference genome or transcriptome and then assigning the reads to specific genes or features. The count values indicate the number of reads that are uniquely mapped to each gene or feature in each sample.

Salmon is already included in ORP for transcript quantification or “counting”. The file needed is labeled as "quant.sf" and can be located in the "quants'' directory of your ORP output for each assembled transcriptome.

The Salmon output file (quant.sf) from ORP does not identify the transcripts, but only preforms quantification as seen in the below example:

![Model](https://github.com/mchunn-1/Bat_Project/blob/main/Screen%20Shot%202023-01-17%20at%202.13.44%20PM.png)

However, ORP does include Diamond as a means of transcript identification within each assembly. The Diamond output file can be found in the "assemblies" directory of your ORP output. The file will be labeled in the format "someName.ORP.diamond.txt" and will look similar to the following:

![Model](https://github.com/mchunn-1/Bat_Project/blob/main/Screen%20Shot%202023-01-17%20at%202.15.14%20PM.png)

Note that while the Salmon out file contains all transcripts within the assembled transcriptome, the Diamond output file contains only those transcripts able to be identified. You will need to merge the two files such that the transcript quantification stats from Salmon are associated with the transcript ID given by Diamond. The merged file should look similar to this:

![Model](https://github.com/mchunn-1/Bat_Project/blob/main/Screen%20Shot%202023-01-17%20at%202.10.51%20PM.png)

From this merged file, you need to generate a “counts file” for each sample that contains only the gene (diamond) ID and the counts (NumReads column). This file should look like this:

![Model](https://github.com/yoheLab/bat_tongues/blob/main/DE%20Analysis/README_images/ID_count_file.png)

This file contains replicates since Diamond reports identification of the same gene in different species. Next you will need to remove these duplicates from the file. The first step in doing so is editing the IDs down to a more readable code. For example, the ID “sp|Q8SPN1|PKR2_BOVIN” needs to be edited to just “PKR2”. This allows you to specify that “PKR2”, regardless of species, should only occur once in your count file. There are 2 scripts provided to 1. Edit the IDs (**geneNamesManipulation.py**)  and 2. Remove the replicates (**removeReplicates.py**). 

Once you have your edited counts files for all of your samples, you need to merge them into one concise count matrix. The **mergeCounts.py** script will do that for you. You should now have a file that contains read counts for each gene across your samples. It will look like this:

![Model](https://github.com/yoheLab/bat_tongues/blob/main/DE%20Analysis/README_images/mergedCounts.png)

The next step is to normalize the counts in the count matrix file. That can done with the **tongueNormalization.R** script. The normalized output file can then be used for DE analysis. Use the **deNorm.R** script as an example on how to run DESeq2 and plot the results in a heat map. 

